lib/elixir/lib/calendar/naive_datetime.ex

```diff
@@ -208,17 +210,33 @@ defmodule NaiveDateTime do
           Calendar.hour(),
           Calendar.minute(),
           Calendar.second(),
-          Calendar.microsecond(),
+          Calendar.microsecond() | 0..999_999,
           Calendar.calendar()
         ) :: {:ok, t} | {:error, atom}
   def new(year, month, day, hour, minute, second, microsecond \\ {0, 0}, calendar \\ Calendar.ISO)
 
   def new(year, month, day, hour, minute, second, microsecond, calendar)
-      when is_integer(microsecond) do
+      when microsecond in 0..999_999 do
     new(year, month, day, hour, minute, second, {microsecond, 6}, calendar)
   end
 
-  def new(year, month, day, hour, minute, second, microsecond, calendar) do
+  def new(
+        year,
+        month,
+        day,
+        hour,
+        minute,
+        second,
+        {microsecond_value, microsecond_precision} = microsecond,
+        calendar
+      )
+      when is_integer(year) and year >= 0 and
+             is_integer(minute) and minute >= 0 and
+             is_integer(second) and second >= 0 and
+             is_integer(hour) and hour >= 0 and
+             is_integer(minute) and minute >= 0 and
+             is_integer(second) and second >= 0 and
+             microsecond_value in 0..999_999 and microsecond_precision in 0..6 do
```

Comment:
Please revert. All of these limitations apply to the Calendar.ISO, it may not apply to others. Probably best to not do changes to this file in this PR.

Reply:
We need to review the specs then, because they are refering to Calendar types